OrbitCore/UserDefinedCaptureData.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "UserDefinedCaptureData.h"
+
+#include "OrbitClientData/FunctionUtils.h"
+
+void UserDefinedCaptureData::InsertFrameTrack(const orbit_client_protos::FunctionInfo& function) {
+  if (frame_track_functions_.count(function) > 0) {
+    ERROR("Trying to add duplicate frame track for function: %s",
+          FunctionUtils::GetDisplayName(function));
+    return;
+  }
+  frame_track_functions_.insert(function);
+}
+
+void UserDefinedCaptureData::EraseFrameTrack(const orbit_client_protos::FunctionInfo& function) {
+  if (!ContainsFrameTrack(function)) {
+    ERROR("Trying to remove non-existent frame track for function: %s",
+          FunctionUtils::GetDisplayName(function));
+    return;
+  }
+  frame_track_functions_.erase(function);
+}
+
+[[nodiscard]] bool UserDefinedCaptureData::ContainsFrameTrack(
+    const orbit_client_protos::FunctionInfo& function) const {
+  return frame_track_functions_.count(function) > 0;
```

Comment:
FunctionSet should be an absl set. This offers a contains method, which I would use here.

Reply:
Done.